java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -4973,6 +4973,218 @@ public void xgroupCreate_xgroupDestroy(BaseClient client) {
+            assertEquals(
+                    null, group1Info.get("lag")); // Gets set to None when we change the last delivered ID
+
+            // Verify xgroup_set_id with entries_read_id effects the returned value from xinfo_groups
+            assertEquals(OK, client.xgroupSetId(key, groupName1, streamId1_1, 1L).get());
+            groups = client.xinfoGroups(key).get();
+            assertEquals(1, groups.length);
+            group1Info = groups[0];
+            assertEquals(groupName1, group1Info.get("name"));
+            assertEquals(2L, group1Info.get("consumers"));
+            assertEquals(3L, group1Info.get("pending"));
+            assertEquals(streamId1_1, group1Info.get("last-delivered-id"));
+            assertEquals(1L, group1Info.get("entries-read"));
+            assertEquals(3L, group1Info.get("lag"));
+        }
+
+        // Add one more consumer group
+        assertEquals(OK, client.xgroupCreate(key, groupName2, streamId0_0).get());
+
+        // Verify that xinfo_groups contains info for 2 consumer groups now
+        groups = client.xinfoGroups(key).get();
+        assertEquals(2, groups.length);
+    }
+
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void xinfoGroupsWithXinfoConsumersAndEdgeCases(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        String stringKey = UUID.randomUUID().toString();
+        String nonExistentKey = UUID.randomUUID().toString();
+        String groupName = "group1" + UUID.randomUUID();
+        String streamId1_0 = "1-0";
+
+        // Passing a non-existing key raises an error
+        ExecutionException executionException =
+                assertThrows(ExecutionException.class, () -> client.xinfoGroups(nonExistentKey).get());
+        assertInstanceOf(RequestException.class, executionException.getCause());
+
+        executionException =
+                assertThrows(
+                        ExecutionException.class, () -> client.xinfoConsumers(nonExistentKey, groupName).get());
+        assertInstanceOf(RequestException.class, executionException.getCause());
+
+        assertEquals(
+                streamId1_0,
+                client
+                        .xadd(
+                                key,
+                                Map.of("f1", "v1", "f2", "v2"),
+                                StreamAddOptions.builder().id(streamId1_0).build())
+                        .get());
+
+        // Passing a non-existing group raises an error
+        executionException =
+                assertThrows(
+                        ExecutionException.class, () -> client.xinfoConsumers(key, "nonExistentGroup").get());
+        assertInstanceOf(RequestException.class, executionException.getCause());
+
+        // No groups exist yet
+        assertDeepEquals(new Map[] {}, client.xinfoGroups(key).get());
```

Comment:
Use typed maps or simplify this
```suggestion
        assertEquals(0, client.xinfoGroups(key).get().size());
```